👋🏻 @geriux. I believe we once discussed the following error when running `avdmanager` commands. 

```
Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
        at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
        at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
        at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
        at com.android.sdklib.tool.AvdManagerCli.run(AvdManagerCli.java:213)
        at com.android.sdklib.tool.AvdManagerCli.main(AvdManagerCli.java:200)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        ... 5 more
```

I was able to resolve this error on my local machine by adding the following to my path _before_ my existing `/tools` addition, [sourced from SO](https://askubuntu.com/a/1388761). 

```diff
diff --git a/zshrc b/zshrc
index be83a92..d1ebb77 100644
--- a/zshrc.local
+++ b/zshrc.local
@@ -36,6 +36,7 @@ export TERM="xterm-256color-italic"
 # Set Android SDK for React Native development
 export ANDROID_HOME=$HOME/Library/Android/sdk
 export PATH=$PATH:$ANDROID_HOME/emulator
+export PATH=$PATH:$ANDROID_HOME/cmdline-tools/latest/bin
 export PATH=$PATH:$ANDROID_HOME/tools
 export PATH=$PATH:$ANDROID_HOME/tools/bin
 export PATH=$PATH:$ANDROID_HOME/platform-tools
```

--- a/packages/react-native-editor/bin/test-e2e-setup.sh
+++ b/packages/react-native-editor/bin/test-e2e-setup.sh
@@ -24,14 +24,14 @@ function log_error() {
 output=$($APPIUM_CMD driver list --installed --json)
 
 if echo "$output" | grep -q 'uiautomator2'; then
-	log_info "UiAutomator2 is available."
+	log_info "UiAutomator2 available."
 else
 	log_info "UiAutomator2 not found, installing..."
 	$APPIUM_CMD driver install uiautomator2
 fi
 
 if echo "$output" | grep -q 'xcuitest'; then
-	log_info "XCUITest is available."
+	log_info "XCUITest available."
 else
 	log_info "XCUITest not found, installing..."
 	$APPIUM_CMD driver install xcuitest
@@ -54,7 +54,7 @@ function detect_or_create_simulator() {
 	local simulators=$(xcrun simctl list devices -j | jq -r --arg runtime "$runtime_name" '.devices | to_entries[] | select(.key | contains($runtime)) | .value[] | .name + "," + .udid')
 
 	if ! echo "$simulators" | grep -q "$simulator_name"; then
-		log_info "$simulator_name ($runtime_name_display) not available, creating..."
+		log_info "$simulator_name ($runtime_name_display) not found, creating..."
 		xcrun simctl create "$simulator_name" "$simulator_name" "com.apple.CoreSimulator.SimRuntime.$runtime_name" > /dev/null
 		log_success "$simulator_name ($runtime_name_display) created."
 	else
@@ -87,7 +87,7 @@ function detect_or_create_emulator() {
 	local emulator=$(emulator -list-avds | grep "$emulator_id")
 
 	if [[ -z $emulator ]]; then
-		log_info "$emulator_name not available, creating..."
+		log_info "$emulator_name not found, creating..."
 		avdmanager create avd -n "$emulator_id" -k "system-images;android-$runtime_api;google_apis;arm64-v8a" -d "$device_id" > /dev/null
 		log_success "$emulator_name created."
 	else

--- a/packages/react-native-editor/bin/test-e2e-setup.sh
+++ b/packages/react-native-editor/bin/test-e2e-setup.sh
@@ -82,11 +82,13 @@ function detect_or_create_emulator() {
 
 	local emulator_name=$1
 	local emulator_id=$(echo "$emulator_name" | sed 's/ /_/g; s/\./_/g')
+	local device_id=$(echo "$emulator_id" | awk -F '_' '{print tolower($1)"_"tolower($2)"_"tolower($3)}')
+	local runtime_api=$(echo "$emulator_id" | awk -F '_' '{print $NF}')
 	local emulator=$(emulator -list-avds | grep "$emulator_id")
 
 	if [[ -z $emulator ]]; then
 		log_info "$emulator_name not available, creating..."
-		avdmanager create avd -n "$emulator_id" -k "system-images;android-30;google_apis;arm64-v8a" -d "pixel_3_xl" > /dev/null
+		avdmanager create avd -n "$emulator_id" -k "system-images;android-$runtime_api;google_apis;arm64-v8a" -d "$device_id" > /dev/null
 		log_success "$emulator_name created."
 	else
 		log_info "$emulator_name available."

